sensors/sqs_service_sensor.py

```diff
@@ -0,0 +1,155 @@
+"""
+This is generic SQS Sensor using boto3 api to fetch messages from sqs queue.
+After receiving a message it's content is passed as payload to a trigger 'aws.sqs_new_message'
+This sensor can be configured either by using config.yaml within a pack or by creating
+following values in datastore:
+    - aws.input_queues (list queues as comma separated string: first_queue,second_queue)
+    - aws.aws_access_key_id
+    - aws.aws_secret_access_key
+    - aws.region
+    - aws.max_number_of_messages (must be between 1 - 10)
+For configuration in config.yaml with config like this
+    setup:
+      aws_access_key_id:
+      aws_access_key_id:
+      region:
+    sqs_sensor:
+      input_queues:
+        - first_queue
+        - second_queue
+    sqs_other:
+        max_number_of_messages: 1
+If any value exist in datastore it will be taken instead of any value in config.yaml
+"""
+
+import six
+import json
+from boto3.session import Session
+from botocore.exceptions import ClientError
+from botocore.exceptions import NoRegionError
+from botocore.exceptions import NoCredentialsError
+from botocore.exceptions import EndpointConnectionError
+
+from st2reactor.sensor.base import Sensor
+
+
+class AWSSQSServiceSensor(Sensor):
+    def __init__(self, sensor_service, config=None):
+        super(AWSSQSServiceSensor, self).__init__(sensor_service=sensor_service, config=config)
+
+    def setup(self):
+        self._logger = self._sensor_service.get_logger(name=self.__class__.__name__)
+
+        self.session = None
+        self.sqs_res = None
+
+    def run(self):
+        # setting SQS ServiceResource object from the parameter of datastore or configuration file
+        self._may_setup_sqs()
+
+        while True:
```

Comment:
I assume there is no yielding needed in this function (aka ``eventlet.sleep(0.01)`` at the end or similar) because ``_receive_messages`` performs a network operation which already needs to yield at some point even if there are no messages to be retrieved.

Otherwise if that's not the case and ``_receive_messages`` could immediately return this could cause CPU spikes and 100% CPU utilization by the sensor process since there is no yielding.

Reply:
boto3 `receive_messages` accepts a `WaitTimeSeconds` argument, which `_receive_messages` defaults to 2 seconds. That's what's keeping the loop from spinning too fast.